capstone/src/main/java/com/google/sps/servlets/DashboardHandlerServlet.java

```diff
@@ -28,19 +31,47 @@
 @WebServlet("/dashboard")
 public class DashboardHandlerServlet extends HttpServlet {
 
+    private static final String TYPE_REQUEST_PROPERTY = "type";
+    private static final String PERSONAL_PROPERTY_TYPE = "personal";
+    private static final String FOLLOWED_PROPERTY_TYPE = "followed";
+    private static final String RECOMMENDED_PROPERTY_TYPE = "recommended";
+
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        
         Key userKey = ((User) request.getAttribute(User.USER_ENTITY_NAME)).getUserKey();
         Filter roomFilter = new FilterPredicate(Room.HOST_PROPERTY_KEY, FilterOperator.EQUAL, userKey);
-        Query query = new Query(Room.ROOM_ENTITY_NAME).setFilter(roomFilter);
-        PreparedQuery results = datastore.prepare(query);
-    
-        ArrayList<Room> rooms = new ArrayList<>();
-        for(Entity entity : results.asIterable()){
-            rooms.add(new Room(entity));
+        PreparedQuery personalResults = datastore.prepare(new Query(Room.ROOM_ENTITY_NAME).setFilter(roomFilter));
+        List<Room> personal = new ArrayList<>();    
```

Comment:
(optional) Always keep variable names as informative as possible. "personal" could be expanded to "personallyFollowedRooms". Careful naming of variables pays off much more than the few saved characters typed